elastic/kibana
# [Lens][LogsDB] Add test suite

## Summary

Test suite for LogsDB in Kibana

Adding tests for the following scenarios:
* [x] All Lens operations (aggs) should work fine with a LogsDB stream
* Mixed index scenarios
  * [x] LogsDB stream (upgraded from regular stream)
  * [x] LogsDB stream without `host.name` field in the mapping
  * [x] LogsDB stream (upgraded from regular stream) + regular index
  * [x] LogsDB stream (upgraded from regular stream) + TSDB stream
  * [x] LogsDB stream (upgraded from regular stream) + another LogsDB stream
  * [x] LogsDB stream (upgraded from regular stream) + TSDB stream downsampled
  * [x] Data stream (downgraded from LogsDB stream)
  * [x] Data stream without `host.name` field in the mapping
  * [x] Data stream (downgraded from LogsDB stream) + regular index
  * [x] Data stream (downgraded from LogsDB stream) + TSDB stream
  * [x] Data stream (downgraded from LogsDB stream) + another LogsDB stream
  * [x] Data stream (downgraded from LogsDB stream) + TSDB stream downsampled

For the mixed scenario suite the following tests are performed:
* Create a date histogram vs `count(bytes)`
  * same test but using the `utc_time` field rather than `@timestamp`
* Create a chart with an annotation layer
  * same test but using the `utc_time` for both data and annotation layer
  * use the `host.name`, `utc_time`, `@timestamp` as extra field in the annotation tooltip
* Create a visualization with ES|QL using `STATS`
